src/search-and-replace-dialog.jsx

```diff
@@ -0,0 +1,286 @@
+/** @jsx React.DOM */
+
+var React = require('react');
+var Util = require("./util.js");
+
+var SearchAndReplaceDialog = React.createClass({
+
+    getDefaultProps: function() {
+        return {
+            onChange() {}    // TODO: throw an error
```

Comment:
None of the rest of our code uses es6 methods; could we use old es5 objects/function() for now?

I'm not sure what we'll eventually do here--es6 methods might be nice, but they also break the greppability of `updateSearchString[=:]` to find the definition of something, which is pretty valuable.